### ISSUE TITLE
Inputs used by build tool to generate resource files misreported as unhandled

### ISSUE THREAD

**Issue by dabrahams** [apple/swift-package-manager#6935]
### Description

Check out the `misreported-inputs` branch of [this minimal project](https://github.com/dabrahams/TestGeneration/tree/misreported-inputs) and run `swift test`

### Expected behavior

No unhandled file warnings from SPM. The two files listed are clearly listed in the inputs of the build tool.

### Actual behavior

**warning: 'testgeneration': found 2 file(s) which are unhandled; explicitly declare them as resources or exclude from the target**
> **/Users/dave/src/TestGeneration/Sources/LibWithResource/Resources/Test2.in
>  /Users/dave/src/TestGeneration/Sources/LibWithResource/Resources/Test1.in**
Building for debugging...
[1/3] Emitting module GenerateResource
[2/3] Compiling GenerateResource GenerateResource.swift
[2/3] Linking GenerateResource
Build complete! (2.60s)
[1/1] Compiling plugin ResourceGeneratorPlugin
Building for debugging...
[1/7] Creating resource files
[2/7] Copying Test2.in
[3/7] Copying Test1.in
[5/12] Compiling LibWithResource LibWithResource.swift
[6/12] Emitting module LibWithResource
[7/12] Compiling LibWithResource resource_bundle_accessor.swift
[8/12] Compiling GenerateResource GenerateResource.swift
[9/12] Emitting module GenerateResource
[9/16] Linking GenerateResource
[11/16] Emitting module AppWithResource
[12/16] Compiling AppWithResource AppWithResource.swift
[12/16] Linking AppWithResource
[14/16] Emitting module ResourceGenerationTests
[15/16] Compiling ResourceGenerationTests ResourceGenerationTests.swift
[15/16] Linking ResourceGenerationPackageTests
Build complete! (15.30s)
Test Suite 'All tests' started at 2023-09-20 14:13:42.378.
Test Suite 'ResourceGenerationPackageTests.xctest' started at 2023-09-20 14:13:42.379.
Test Suite 'ResourceGenerationTests' started at 2023-09-20 14:13:42.379.
Test Case '-[ResourceGenerationTests.ResourceGenerationTests testIt]' started.
Test Case '-[ResourceGenerationTests.ResourceGenerationTests testIt]' passed (0.002 seconds).
Test Suite 'ResourceGenerationTests' passed at 2023-09-20 14:13:42.381.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.002 (0.002) seconds
Test Suite 'ResourceGenerationPackageTests.xctest' passed at 2023-09-20 14:13:42.381.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.002 (0.002) seconds
Test Suite 'All tests' passed at 2023-09-20 14:13:42.381.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.002 (0.003) seconds

### Swift Package Manager version/commit hash

Swift Package Manager - Swift 5.9.0

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx13.0
Darwin davea-mbp16.corp.adobe.com 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000 arm64

**Comment by neonichu** [comment]
https://github.com/apple/swift-package-manager/pull/6777 will fix this.

**Comment by dabrahams** [comment]
@neonichu according to your description

> We're emitting warnings for unhandled files during a build, but weren't taking into account cases where only a subset gets built.

it doesn't sound like the same problem.  _Everything_ is getting built in my case.  Did you try to reproduce the problem (it's easy!) with #6777 applied?

**Comment by neonichu** [comment]
You can see from the output that two builds are happening:

```
[...]
Building for debugging...
[1/3] Emitting module GenerateResource
[2/3] Compiling GenerateResource GenerateResource.swift
[2/3] Linking GenerateResource
Build complete! (2.60s)
[1/1] Compiling plugin ResourceGeneratorPlugin
Building for debugging...
[...]
```

The first one is the subset build which emits the warnings.

**Comment by dabrahams** [comment]
OK.  Just out of curiosity, why does `swift test` build a subset?

**Comment by rauhul** [comment]
Wow this is really annoying; using `exclude: [<files>]` to silence the warning, breaks the plugin by hiding the source file from the plugin 😞

**Comment by dabrahams** [comment]
> Wow this is really annoying; using exclude: [<files>] to silence the warning, breaks the plugin by hiding the source file from the plugin 😞

Yes, but there are almost invariably [other ways](https://github.com/dabrahams/SPMBuildToolSupport/commit/50d6f1d52c50c0d26abcec53d9f6bc68177e1f8e) to find the sources, if you have control over the plugin source.

**Comment by rauhul** [comment]
> > Wow this is really annoying; using exclude: [<files>] to silence the warning, breaks the plugin by hiding the source file from the plugin 😞
> 
> 
> 
> Yes, but there are almost invariably [other ways](https://github.com/dabrahams/SPMBuildToolSupport/commit/50d6f1d52c50c0d26abcec53d9f6bc68177e1f8e) to find the sources, if you have control over the plugin source.

Ah that's a great, but undeniably gross solution, lol. I will copy it!

**Comment by rauhul** [comment]
@MaxDesiatov @pusukuri was this resolved by #7300?
